src/app/cli/src/init/client.ml

```diff
@@ -851,15 +858,17 @@ let set_snark_worker =
   in
   Command.async
     ~summary:"Set key you wish to snark work with or disable snark working"
-    (Cli_lib.Background_daemon.init ~rest:true public_key_flag
-       ~f:(fun port optional_public_key ->
-         let open Graphql_client in
+    (Cli_lib.Background_daemon.graphql_init public_key_flag
+       ~f:(fun (module Client) optional_public_key ->
```

Comment:
agree with @Schmavery here, I think it's simpler to pass simple values (an integer) rather than first class modules (so if the only reason you used the first class module was for the port parameter, probably best to move back to the port arg)